docs/CircleCI.md

```diff
@@ -0,0 +1,66 @@
+- `parameters`: Reusable keys that are used within `step` definitions within a `job`. Default parameters that are specified in a `parameters` block are passed through into the [command step](https://buildkite.com/docs/pipelines/command-step)'s `commands` if specified.
+- `steps`: A collection of commands that are executed as part of a CircleCI `job`. Steps can be defined within an `alias`.
+
+#### Docker
+
+- `docker`: Specifies that the `job` will run within a Docker container (by its `image` property) with the use of the [Docker Buildkite Plugin](https://github.com/buildkite-plugins/docker-buildkite-plugin). Additionally, the [Docker Login Plugin](https://github.com/buildkite-plugins/docker-login-buildkite-plugin) is appended if an `auth` property is defined, or the [ECR Buildkite Plugin](https://github.com/buildkite-plugins/ecr-buildkite-plugin) if an `aws-auth` property is defined within the `docker` property. Sets the [agent targeting](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) for the generated [command step](https://buildkite.com/docs/pipelines/command-step) to `executor_type: docker`.
+
+#### Machine 
+
+- `machine`: Specifies that the `job` will run on a machine execution environment. The `resource_class` parameter will be utilised to specify [agent targeting](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) for the generated [command step](https://buildkite.com/docs/pipelines/command-step), in addition to setting the additional tags of ``executor_type: machine` and `executor_image: self-hosted`.
+- `resource_class`: The specification of compute that the executor will require in running a job. This is used to specify the `resource_class` [agent targeting](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) tag for the corresponding [command step](https://buildkite.com/docs/pipelines/command-step).
+
+#### macOS 
+
+- `macos`: Specifies that the `job` will run on a macOS bases execution environment. The [agent targeting](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) tags for the generated [command step](https://buildkite.com/docs/pipelines/command-step) will be set to `executor_type: osx`, as well as the specified version of `xcode` from the `macos` parameters as `executor_xcode: <version>`.
+
+#### Windows
+
+- `windows`: Specifies that the `job` will run on a Windows based execution environment. The [agent targeting](https://buildkite.com/docs/pipelines/defining-steps#targeting-specific-agents) tags for the generated [command step](https://buildkite.com/docs/pipelines/command-step) will be set to `executor_type: windows`.
+
+Currently, these following `job` parameters are not supported by the Buildkite Migration tool:
+
+- `executors`: The `executor` key is currently not supported yet (Supported keys for `job` execution environments are `docker`, `machine`, `macos` and `windows`).
```

Comment:
This is quite confusing as you just specified how each of the values would be translated.

The reality is that there is no translation for this key because of the fact that the execution environment in buildkite is defined by the agent it will run on. To allow for different configurations, agent tags are used (but queues are also available).

Reply:
I revised this in the latest commit - to more state that the actual ex environment is determined by a combination of queue/tag config that is setup for specific agents